### PR TITLE
DateComponents: Fix Overlapping accesses to 'self' warnings.

### PR DESCRIPTION
**Pull request by spevans**

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci please test

File: Foundation/DateComponents.swift [hunk 1]
```diff
@@ -217,8 +259,9 @@ public struct DateComponents : ReferenceConvertible, Hashable, Equatable, _Mutab
     ///
     /// The calendar and timeZone and isLeapMonth properties cannot be set by this method.
     public mutating func setValue(_ value: Int?, for component: Calendar.Component) {
+        let _value = _setter(value)
```
Comment (ianpartridge) [inline Foundation/DateComponents.swift]: Was it intentional that this is `_value` instead of `value`? All the others were `value`.

**Comment by parkera** [comment]
I believe this code is shared with the overlay.

**Comment by ianpartridge** [comment]
@spevans can you open a PR against https://github.com/apple/swift/blob/master/stdlib/public/SDK/Foundation/DateComponents.swift too?

@swift-ci please test and merge

**Pull request merged**